File: backoffice/src/diningTable/domain/events/DiningTableNumberChanged.py
"""
 *
 * Libraries 
 *
"""

from ..valueObjects.DiningTableNumber import DiningTableNumber
from ..valueObjects.DiningTableId     import DiningTableId
from src.shared.domain                import DomainEvent

"""
 *
 * Class
 *
"""

class DiningTableNumberChanged( DomainEvent ):

    """
     *
     * Attributes 
     *
    """

    __id     : DiningTableId
    __number : DiningTableNumber

    """
     *
     * Methods 
     *
    """

    def __init__( 
        self, 
        id        : DiningTableId     = None, 
        number    : DiningTableNumber = None,
        eventId   : str               = None,
        timestamp : int               = None
    ) -> None:
        super().__init__( eventId, timestamp )
        self.__id     = id
        self.__number = number
    
    def getId( self ) -> DiningTableId:
        return self.__id
    
    def getNumber( self ) -> DiningTableNumber:
        return self.__number
    
    def getEventName( self ) -> str:
        return 'dining.table.number.changed'
    
    def toPrimitives( self ) -> dict[str, str | int]:
        # Variables
        data : dict[str, str | int]
        # Code
        data = {
            'id'     : self.__id.getValue(),
            'number' : self.__number.getValue()
        }
        return data

    def fromPrimitives( self, 
        eventId   : str, 
        timestamp : int, 
        data      : dict[str, str | int]
    ) -> object:
        # Variables
        domainEvent : object
        # Code
        domainEvent = DiningTableNumberChanged(
            id        = DiningTableId( data['id'] ),
            number    = DiningTableNumber( data['number'] ),
            eventId   = eventId,
            timestamp = timestamp
        )
        return domainEvent

File: preparation/src/diningTable/domain/events/DiningTableNumberChanged.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class DiningTableNumberChanged : DomainEvent {

        /* 
         * 
         * Attributes
         *
        */

        private DiningTableId     id;
        private DiningTableNumber number;

        /* 
         * 
         * Methods
         *
        */

        public DiningTableNumberChanged() {
        }

        public DiningTableNumberChanged(
            DiningTableId     id,
            DiningTableNumber number
        ) {
            this.id     = id;
            this.number = number;
        }

        public DiningTableNumberChanged(
            DiningTableId     id,
            DiningTableNumber number,
            string            eventId,
            int               timestamp
        ) : base( eventId, timestamp ) {
            this.id     = id;
            this.number = number;
        }

        public DiningTableId getId() {
            return id;
        }

        public DiningTableNumber getNumber() {
            return number;
        }

        public override DomainEvent fromPrimitives(
            string                     eventId, 
            int                        timestamp, 
            Dictionary<string, object> data
        ) {
            // Variables
            DiningTableNumberChanged domainEvent;
            // Code
            domainEvent = new DiningTableNumberChanged(
                new DiningTableId( data["id"].ToString() ),
                new DiningTableNumber( Int32.Parse( data["number"].ToString() ) ),
                eventId,
                timestamp
            );
            return domainEvent;
        }

        public override string getEventName() {
            return "dining.table.number.changed";
        }

        public override Dictionary<string, object> toPrimitives() {
            // Variables
            Dictionary<string, object> data;
            // Code
            data = new Dictionary<string, object>();
            data.Add( "id", id.getValue() );
            data.Add( "number", "" + number.getValue() );
            return data;
        }

    }

}